manifests/init.pp

```diff
@@ -100,7 +100,9 @@
   Optional[String] $systemd_beat_log_opts_override                    = undef,
   String $systemd_beat_log_opts_template                              = $filebeat::params::systemd_beat_log_opts_template,
   String $systemd_override_dir                                        = $filebeat::params::systemd_override_dir,
-
+  Boolean $chocolatey_provider                                        = $filebeat::params::chocolatey_provider,
```

Comment:
Any thoughts on using a parameter like `package_provider` and default to exec (only for backwards compatibility)? If other methods ever show up we won't need a parameter per method this way.

Of course if the method you suggest is common across other modules then let's stick with what people will see.